pkg/stanza/operator/helper/flusher.go

```diff
@@ -54,9 +55,12 @@ type Flusher struct {
 	// if previousDataLength = 0 - no new data have been received after flush
 	// if previousDataLength > 0 - there is data which has not been flushed yet and it doesn't changed since lastDataChange
 	previousDataLength int
+	rwLock             sync.RWMutex
 }
 
 func (f *Flusher) UpdateDataChangeTime(length int) {
+	f.rwLock.Lock()
+	defer f.rwLock.Unlock()
```

Comment:
We'll have to ensure this doesn't impact performance when the gate is not enabled. If benchmarks can show it's not an issue, that's fine but otherwise can we just check the gate?

Reply:
Sure